files/default/sql/ddl/updates/3.4.0.sql

```diff
@@ -158,6 +158,23 @@ ALTER TABLE `hopsworks`.`feature_group` ADD COLUMN `topic_name` VARCHAR(255) DEF
 ALTER TABLE `hopsworks`.`project` ADD COLUMN `topic_name` VARCHAR(255) DEFAULT NULL;
 
 SET SQL_SAFE_UPDATES = 0;
+-- deletes unused topics (fg was deleted)
+DELETE
+FROM `hopsworks`.`project_topics`
+WHERE REGEXP_SUBSTR(`topic_name`, "^([0-9]+)_([0-9]+)_(.+)_([0-9]+)(_onlinefs|$)")
+  AND REGEXP_REPLACE(`topic_name`, "^([0-9]+)_([0-9]+)_(.+)_([0-9]+)(_onlinefs|$)", '$2') NOT IN (SELECT `feature_group`.`id` FROM `feature_group`);
+
+-- remove the foreign key to subject from data ingestion topics (henceforth not needed)
+UPDATE `hopsworks`.`project_topics`
```

Comment:
I'm not really sure why do we need to remove the fk in here

Reply:
technically not necessary to fix upgrade failing, but if not done it could cause problem when deleting online enable fg here: https://github.com/logicalclocks/hopsworks-ee/pull/1544/files (it would throw exception, since we would try to delete subject that is referenced by project_topics)